server/db/index.js

```diff
@@ -60,33 +60,40 @@ class Database {
   async getPlace(placeId) {
     try {
       // Get place
-      let res = await this.pool.query('SELECT * FROM places WHERE id = $1', [
-        placeId,
-      ]);
-      const place = res.rows[0];
-
-      // Get place's list
-      res = await this.pool.query('SELECT * FROM lists where id = $1', [
-        place.listid,
-      ]);
-      const list = res.rows[0];
-
-      // Get place's location
-      res = await this.pool.query('SELECT * FROM locations WHERE id = $1', [
-        placeId,
-      ]);
-      const location = res.rows[0];
-
-      // Get location's map center
-      res = await this.pool.query('SELECT * FROM mapcenters WHERE id = $1', [
-        location.id,
-      ]);
-      const mapCenter = res.rows[0];
+      let res = await this.pool.query(
+        `SELECT places.id AS placeid, places.name AS placename, places.rating, 
+                places.description AS placedescription, locations.name AS locationname, 
+                locations.address, mapcenters.lat, mapcenters.lng, 
+                lists.id AS listid, lists.name AS listname, lists.description AS listdescription 
+         FROM places 
+         LEFT JOIN locations ON places.id = locations.placeid 
```

Comment:
Location and Place and Map Center has 1 to 1 correlation, and a Place must have a Location and must have a Map Center, am I right? In that case, you can just combine the tables together. Joining isn't exactly free either.

For more info about designing tables, see https://www.guru99.com/database-normalization.html as a guide, depending on the tradeoffs, whether speed, or referential consistency. I think majority of database out there only goes to 3NF.

Reply:
Ohh yeahh....wow that was stupid of me...lol